components/datetime/src/datetime.rs

```diff
@@ -23,6 +23,348 @@ use icu_provider::prelude::*;
+    ///
+    /// let datetime = DateTime::new_gregorian_datetime(2020, 9, 1, 12, 34, 28)
+    ///     .expect("Failed to construct DateTime.");
+    ///
+    /// let _ = tf.format_to_string(&datetime);
+    /// ```
+    #[inline]
+    pub fn format_to_string(&self, value: &impl DateTimeInput) -> String {
+        self.0.format_to_string(value)
+    }
+}
+
+/// [`DateFormat`] is a structure of the [`icu_datetime`] component that provides date formatting only.
+/// When constructed, it uses data from the [data provider], selected [`Locale`] and provided preferences to
+/// collect all data necessary to format any date into that locale.
+///
+/// For that reason, one should think of the process of formatting a date in two steps - first, a computational
+/// heavy construction of [`DateFormat`], and then fast formatting of [`DateTimeInput`] data using the instance.
+///
+/// [`icu_datetime`]: crate
+/// [`DateTimeFormat`]: crate::datetime::DateFormat
+///
+/// # Examples
+///
+/// ```
+/// use icu::calendar::{DateTime, Gregorian};
+/// use icu::datetime::{DateFormat, options::length::Date};
+/// use icu::locid::locale;
+/// use icu_provider::inv::InvariantDataProvider;
+///
+/// let provider = InvariantDataProvider;
+///
+/// let df = DateFormat::<Gregorian>::try_new(locale!("en"), &provider, Date::Full)
+///     .expect("Failed to create DateFormat instance.");
+///
+/// let datetime = DateTime::new_gregorian_datetime(2020, 9, 1, 12, 34, 28)
+///     .expect("Failed to construct DateTime.");
+///
+/// let value = df.format_to_string(&datetime);
+/// ```
+///
+/// This model replicates that of `ICU` and `ECMA402`.
+///
+/// [data provider]: icu_provider
+pub struct DateFormat<C>(pub(super) raw::DateFormat, PhantomData<C>);
+
+impl<C: CldrCalendar> DateFormat<C> {
+    /// Constructor that takes a selected [`Locale`], reference to a [data provider] and
+    /// a list of options, then collects all data necessary to format date and time values into the given locale.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use icu::calendar::Gregorian;
+    /// use icu::datetime::{DateFormat, options::length::Date};
+    /// use icu::locid::locale;
+    /// use icu_provider::inv::InvariantDataProvider;
+    ///
+    /// let provider = InvariantDataProvider;
+    ///
+    /// let df = DateFormat::<Gregorian>::try_new(locale!("en"), &provider, Date::Full);
+    ///
+    /// assert_eq!(df.is_ok(), true);
+    /// ```
+    ///
+    /// [data provider]: icu_provider
+    #[inline]
+    pub fn try_new<T: Into<Locale>, D>(
+        locale: T,
+        data_provider: &D,
+        length: length::Date,
+    ) -> Result<Self, DateTimeFormatError>
+    where
+        D: ResourceProvider<DateSymbolsV1Marker>
+            + ResourceProvider<DatePatternsV1Marker>
+            + ResourceProvider<DecimalSymbolsV1Marker>
+            + ResourceProvider<OrdinalV1Marker>
+            + ResourceProvider<WeekDataV1Marker>
```

Comment:
Thought: We _may_ be able to relax the Ordinal and WeekData requirements given that I think those patterns are not reachable from date lengths, but this can be a follow-up and is an easy change to make post-1.0 given our decisions regarding constructor signatures.